### ISSUE TITLE
Questions Regarding the  code Implementation

### ISSUE THREAD

**Issue by Yutong-Dai** [ebagdasa/backdoors101#16]
Hi, thanks for the code!

I have some questions regarding the code implementation.

1. In the [line 119](https://github.com/ebagdasa/backdoors101/blob/af12c08280fe59380f74c05e2737eb2e92a80fdf/attack.py#L119)  of `attack.py` file, I think the purpose here is to scale the local update of a compromised client so that the local update can replace the global model as described in the equation (3) of the paper *How To Backdoor Federated Learning*. In the implementation, the scaling factor is set to `self.params.fl_weight_scale`.  And in the config file, [it](https://github.com/ebagdasa/backdoors101/blob/af12c08280fe59380f74c05e2737eb2e92a80fdf/configs/cifar_fed.yaml#L46) was set to the total number of participants, **However**,  I think this is not correct as it does not take the parameter `fl_eta` (server-side stepsize) into account, which is used in [here](https://github.com/ebagdasa/backdoors101/blob/af12c08280fe59380f74c05e2737eb2e92a80fdf/tasks/fl/fl_task.py#L126) to perform global weight update. Also, I think it ignores the fact that the training protocol allows partial participation as implied by this line [here](https://github.com/ebagdasa/backdoors101/blob/af12c08280fe59380f74c05e2737eb2e92a80fdf/training.py#L83). From what I have in mind, the scaling factor should be `num_of_participants_at_the_attacked_round / fl_eta`.

2.  In the model [`simple.py`](https://github.com/ebagdasa/backdoors101/blob/af12c08280fe59380f74c05e2737eb2e92a80fdf/models/simple.py#L32), a `F.log_softmax` is applied. But later, the [attack](https://github.com/ebagdasa/backdoors101/blob/af12c08280fe59380f74c05e2737eb2e92a80fdf/attack.py#L101) uses the the  `nn.CrossEntropyLoss`, which ends with "normalizing" the neural net's output twice. This seems to be a bit weird to me. Is there any specific reason for this?

Thank you!

**Comment by ebagdasa** [comment]
hey @Yutong-Dai yeah, you are right for 1, re 2 I am not sure which one is right, happy to update it too. You are welcome to submit PR or I can try to fix it too
